### ISSUE TITLE
Unexpected Relationship between Range (rated) and Range (est.) in Internal Drives Dashboard, Drive panel

### ISSUE THREAD

**Issue by tjhart** [teslamate-org/teslamate#3763]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

When viewing the 'Drive' panel of the 'Internal Drives' dashboard, the Range (rated) and Range (est.) graph lines relationship to each other is unexpected.

### Expected Behavior

Given that Range (rated) and Range (est.) are measured in the same unit of measurement, I expect values that are close to each other to be graphed close to each other, and values that are relatively far apart to be visually far apart.

### Steps To Reproduce

1. Install Teslamate 1.28.4 (docker implementation)
2. Provide car details for a 2022 Model 3 Tesla (details are probably irrelevant for this issue, but I can provide a partial record on request)
3. import data from `drives.csv` and `drive-2.csv`
4. import data from `positions.csv` and `positions-2.csv`
5. in Grafana, view `drives` dashboard
6. Select drive with time range 2024-03-10 15:39:27.893 UTC to 2024-03-10 17:50:53.855 UTC (`drive-1` ) or time range 2024-03-21 16:30:02 UTC to 2024-03-21 16:40:52 (`drive-2`)
7. Observe unusual relationship between Range (rated) and Range (est.) lines in `Drive` panel (see screen shots)

### Relevant log output

```shell
teslamate-1  | 2024-03-22 14:10:49.587 [info] Migrations already up
teslamate-1  | 2024-03-22 14:10:51.031 [info] No tzdata release files found in custom data dir. Copying release file from tzdata priv dir.
teslamate-1  | 2024-03-22 14:10:51.074 [info] System Info: Erlang/OTP 26 (jit)
teslamate-1  | 2024-03-22 14:10:51.074 [info] Version: 1.28.4
teslamate-1  | 2024-03-22 14:10:52.272 [info] POST https://auth.tesla.com/oauth2/v3/token -> 200 (1189.841 ms)
teslamate-1  | 2024-03-22 14:10:52.272 [info] Refreshed api tokens
teslamate-1  | 2024-03-22 14:10:52.273 [info] Scheduling token refresh in 6 h
teslamate-1  | 2024-03-22 14:10:52.276 [info] Running TeslaMateWeb.Endpoint with cowboy 2.10.0 at :::4000 (http)
teslamate-1  | 2024-03-22 14:10:52.277 [info] Access TeslaMateWeb.Endpoint at http://localhost
teslamate-1  | 2024-03-22 14:10:52.661 [info] Starting logger for 'SA Node'
teslamate-1  | 2024-03-22 14:10:52.667 [info] MQTT connection has been established
teslamate-1  | 2024-03-22 14:10:52.789 car_id=1 [info] Start / :asleep
teslamate-1  | 2024-03-22 14:10:54.191 [info] tzdata release in place is from a file last modified Fri, 22 Oct 2021 02:20:47 GMT. Release file on server was last modified Thu, 01 Feb 2024 18:40:48 GMT.
teslamate-1  | 2024-03-22 14:10:54.759 [info] Tzdata has updated the release from 2021e to 2024a
teslamate-1  | 2024-03-22 14:13:46.431 [info] GET /
teslamate-1  | 2024-03-22 14:13:46.450 [info] Sent 200 in 18ms
teslamate-1  | 2024-03-22 14:53:19.530 [info] GET /
teslamate-1  | 2024-03-22 14:53:19.541 [info] Sent 200 in 10ms
teslamate-1  | 2024-03-22 15:47:37.451 [info] GET /
teslamate-1  | 2024-03-22 15:47:37.460 [info] Sent 200 in 9ms

The system has been updated since the data was recorded, so I have no logs for that time.
```

### Screenshots

Irrelevant graph lines were set to 'transparent' color to aid in demonstration.

Drive 1
Baseline units apart: 16
<img width="946" alt="Screenshot 2024-03-22 at 10 57 22 AM" src="https://github.com/teslamate-org/teslamate/assets/95165/daad2011-fd7c-4e4d-949b-16691cf4a18f">

Units apart: 1
<img width="951" alt="Screenshot 2024-03-22 at 10 57 58 AM" src="https://github.com/teslamate-org/teslamate/assets/95165/deaa7357-e56c-4d8e-ad25-4fc6ebbdb72d">

Units apart: -13
<img width="948" alt="Screenshot 2024-03-22 at 10 57 43 AM" src="https://github.com/teslamate-org/teslamate/assets/95165/18d650f6-fe30-4eda-8810-2a6b2c85be37">

Units apart: -7
<img width="997" alt="Screenshot 2024-03-22 at 11 13 00 AM" src="https://github.com/teslamate-org/teslamate/assets/95165/941061a7-bf0d-422b-9fcc-0987d32f3f08">

Drive 2
Baseline Units apart: -23
<img width="837" alt="Screenshot 2024-03-22 at 3 28 38 PM" src="https://github.com/teslamate-org/teslamate/assets/95165/7b582bf0-cc54-428d-9993-ceaeb7907a92">

Units apart: 0
<img width="833" alt="Screenshot 2024-03-22 at 3 28 53 PM" src="https://github.com/teslamate-org/teslamate/assets/95165/c4851023-f73b-4ba2-a226-f3869d6f6eed">

Units apart: -44
<img width="835" alt="Screenshot 2024-03-22 at 3 29 07 PM" src="https://github.com/teslamate-org/teslamate/assets/95165/946510fe-cae2-4020-b7f3-b3b340951e10">

Units apart: -41
<img width="1110" alt="Screenshot 2024-03-22 at 3 34 06 PM" src="https://github.com/teslamate-org/teslamate/assets/95165/5288e5f6-84a8-4b62-87d1-372c038f8b2f">

### Additional data

[internal-drive-drive.csv](https://github.com/teslamate-org/teslamate/files/14725909/internal-drive-drive.csv)
[positions.csv](https://github.com/teslamate-org/teslamate/files/14725952/positions.csv)
[drives.csv](https://github.com/teslamate-org/teslamate/files/14726214/drives.csv)
[drive-2.csv](https://github.com/teslamate-org/teslamate/files/14728147/drive-2.csv)
[positions-2.csv](https://github.com/teslamate-org/teslamate/files/14728149/positions-2.csv)

### Type of installation

Docker

### Version

v1.28.4

**Comment by JakobLichterfeld** [comment]
Not really sure if I understand your issue correctly. The graph do show that the values sometimes are really close or even the same, but when there is a big change in Range (est) due to hard acceleration or decelleration the cursor does not snag to the right value when you are zoomed out?

**Comment by tjhart** [comment]
@JakobLichterfeld Note the 3 screen shots (all of the exact same graph), the units I calculated, and the displayed difference between the values:

1. As a baseline, The first screen shot shows the initial rated (244 mi) and estimated range (228 mi). The values are 16 units apart, and you can visually see some distance between them on the y axis, suggesting a delta in distance
2. The second screen shot shows rated (235 mi) and estimated range (234 mi). The values are 1 unit apart on the y axis, and (at least to me) a surprising gap between the two, but compare that to the 3rd screenshot:
3. Rated (234 mi) and estimated range (247 mi) they're now -13 units apart on the y axis, but are graphed at almost the exact same coordinate, with the estimated just being slightly **below** the rated. Compared to the baseline, I would expect the estimated coordinate to be above the rated coordinate, by a little less distance than the baseline
4. Rated (125 mi) and estimated (132 mi). They're now -7 units apart, with the estimated coordinate being significantly below the rated coordinate (y axis). Further below than the baseline, when the distance was 16 units. I would expect the estimated coordinate to be **above** the rated coordinate.

* The estimated range and rated range are the same unit (distance).
* At any point the estimated range and rated range were the same (both measured in distance), I expect the graph point to be the same coordinate.
* When the estimated range is higher than the rated range, I would expect the estimated coordinate to be higher than the rated on the y axis. The screen shots demonstrate that is not always the case.
* When the estimated range is lower than the rated range, I would expect the estimated coordinate to be below the rated coordinate. 
* I expect the distance between the estimated and rated ranges y axis value to demonstrate the delta between the two values. It doesn't.

**Comment by tjhart** [comment]
I don't understand "the cursor does not snag to the correct value" or "zoom level is too small for quick changes" What do you mean?

I recognize that in the graph above, Grafana is dealing with multiple records per point on the x axis. However, that does not explain the issue I'm seeing. The displayed values line up with the underlying data. I can't find 'adjacent' data that would explain the above.

Screen shot 3 explains the problem very well when you look at the data. The time is 2024-03-10 15:52:25 UTC. Here are records near that timestamp that have range values:
```
select id,
       date,
       convert_km(p.rated_battery_range_km, 'mi') as range_rated_mi,
       convert_km(p.est_battery_range_km, 'mi')   as range_estimated_mi
from positions p
where p.drive_id = 1781
  and p.date between '2024-03-10 15:52' and '2024-03-10 15:53'
  and p.rated_battery_range_km is not null
order by p.date


   id    |          date           |    range_rated_mi    |  range_estimated_mi  
---------+-------------------------+----------------------+----------------------
 4569786 | 2024-03-10 15:52:14.66  | 233.9903314402177290 | 247.0888687287956554
 4569841 | 2024-03-10 15:52:29.967 | 233.9903314402177290 | 247.0888687287956554
 4569890 | 2024-03-10 15:52:45.26  | 233.9903314402177290 | 246.7036176320727752
```

1. Screen shot 1 represents position ID 4567906. The values displayed in the screen shot for range equal the values displayed. This is the second record for this drive. The previous record had no range values. The next range value doesn't show up for 5 seconds - ID 4567906, date 2024-03-10 15:39:43.57, and the values are effectively the same.
2. Screen shot 2 represents position ID 4569599. The values displayed are correct for this record. The previous record with a range value was at 2024-03-10 15:50:58.281, and the next record with a range value isn't until 2024-03-10 15:51:28.858, ID 4569640. The values before and after are effectively the same
3. Screen shot 3 represents position ID 4569841. The values displayed are correct for this record. The previous record with a range value was at 2024-03-10 15:52:14.66 (id 4569786), and the next was at 2024-03-10 15:52:45.26 (id 4569890), the values before and after are less than 1 unit different.
4. Screen shot 4 represents position ID 4589081. The values displayed are correct for this record. The previous record with a range value was at 2024-03-10 17:47:50.972 (id 4589026), and the next  was at 2024-03-10 17:48:21.738 (id 4589137). The values before and after are effectively the same

**Comment by tjhart** [comment]
@JakobLichterfeld I've provided data for a second drive which still demonstrates the same problem. This drive is only 10 minutes long, but the same issues are demonstrated.

**Comment by cwanja** [comment]
I believe what is happening is that these values are not plotted against the Y axis with the values we had displayed. And thus they do not "fluctuate" how you would expect them too on a normal graph (e.g. proper spacing). Notice that on the left, there is no value for the range values in the axis - it is speed and power. I think Grafana is just doing its best to plot the range values. @DrMichael you see anything different to this? 

I can make them show, but look how confusing the right hand side is:
![Screenshot 2024-03-22 161611](https://github.com/teslamate-org/teslamate/assets/54645561/f3b9cb3e-1cfd-4967-bc91-b9df48dcb110)

If I have my understanding correctly then, I am seeing the same behavior @tjhart - see screenshot and where my tooltip is. 
![Screenshot 2024-03-22 161727](https://github.com/teslamate-org/teslamate/assets/54645561/aa4c3750-7815-4291-aa1e-31c4e101fb37)

But I believe if we add them, it is going to be too confusing.

**Comment by tjhart** [comment]
@JakobLichterfeld I don't have experience with Grafana, so I'm not sure if this is the correct solution, but it resolves my issue. 

I've fiddled with the pane definition for 'Internal Drive -> Drives'. Override 12 is defined as:
```
Fields with name matching regex: range_estimated_.*
Standard options > Display name: Range (est.)
```

Which is different than Override 9:

```
Fields with name matching regex: range_rated_.*
Standard options > Display name: Range (rated)
Standard options > Min: 0
```

Removing `Standard options > Min: 0` from both, or setting it to the same on both, gives me what I expect to see on the graph.

**Comment by tjhart** [comment]
@cwanja I understand you're trying to represent a LOT of data on this graph, and space is a premium. Dunno what the right solution is for everyone.

I do expect lines of the same unit of measure should sync, whether the labels are displayed or not. I'm ok with not seeing a 'miles' label. The popup for values makes it clear enough for me.

**Comment by cwanja** [comment]
@tjhart definitely take a look at it later, my gut tells me it was oversight. I know these graphs have been a bit pain even when Adrian was around. Would want @DrMichael to validate further as well.

**Comment by tjhart** [comment]
@cwanja has this issue been validated? I can create a MR with the field Min set to 0, but I'd prefer to know that it'd be accepted relatively soon.

**Comment by cwanja** [comment]
@DrMichael @Dulanic - can you review this thread and the proposed change below. To resolve the irregular lines spacing in the Drive Details panel, @tjhart proposed one of the following fixes (note the **orange** and **red** lines):
1. Add `Standard options > Min 0` to Override 12
<img width="1037" alt="Screenshot 2024-04-15 at 9 37 25 PM" src="https://github.com/teslamate-org/teslamate/assets/54645561/ac30609c-864e-4718-a19b-855f1612e51f">

2. Remove `Standard options > Min 0` from Override 9
<img width="1033" alt="Screenshot 2024-04-15 at 9 37 38 PM" src="https://github.com/teslamate-org/teslamate/assets/54645561/2ee4a820-a344-44d1-b7ff-af95ae89cbad">

In my testing, **option two** looks like the better path. 
1. Do you agree? 
2. Do you see / know any downstream impacts by removing this override property? 

@jheredianet would appreciate any insights you have as well.

**Comment by cwanja** [comment]
If no or no negative feedback is given in 48 hours, I will propose a PR **removing** the property on override 9.

**Comment by jheredianet** [comment]
> @jheredianet would appreciate any insights you have as well.

Hi @cwanja @tjhart, before any change, please could you test with this PR? (already merged a few days ago) https://github.com/teslamate-org/teslamate/pull/3836

I've already changed the axis, because it's true that the info is a little confusing since it shows many values with different value metrics of different units, so It's hard for Grafana to mix them all. So, what I did was to put the "power" to the right and the rest on the left (as they always were), also updated the color patterns, and the most important change is the "center cero" axis, with it the y axis could visualize better the regen on a drive, on the other hand the ranges (estimate, rated or ideal depending on the user preference) show as expected. 

New view with all metrics:
![image](https://github.com/teslamate-org/teslamate/assets/45735191/42a3d1ee-4eed-4c07-99dc-2e068c07fae5)


Only Speed, Range rated and ideal selected:
 
![image](https://github.com/teslamate-org/teslamate/assets/45735191/19c0b3cb-0e74-4028-92f0-c82319acfe4f)

Could you check if it is ok or is it necessary to Remove Standard options > Min 0?

**Comment by tjhart** [comment]
@jheredianet and @cwanja I've reviewed the PR as it relates to this issue. The images above, and reviewing my own data with the changes, do NOT resolve the issue I see.

Note in this image ![](https://private-user-images.githubusercontent.com/45735191/322739047-19c0b3cb-0e74-4028-92f0-c82319acfe4f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTMyODMyOTIsIm5iZiI6MTcxMzI4Mjk5MiwicGF0aCI6Ii80NTczNTE5MS8zMjI3MzkwNDctMTljMGIzY2ItMGU3NC00MDI4LTkyZjAtYzgyMzE5YWNmZTRmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MTYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDE2VDE1NTYzMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWNkMjg3MmQ1ZDhiZjljOTFhNWY1YzdkYmUzY2FjOWU1NTE5YjQ2ZWUzNDlkZDdiYzVmOGJkZDg0ZjY4OTM0YTcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.Xco8SIZIMap31NcCqUWl3UIqwDqewMQPMjNJUfnCP0s)

The rated range has a higher value, but the estimated range is higher in the displayed graph.

I displayed the data range on the right axis, (only for the purposes of testing and inspection), and to my surprise the scale for the estimated range dipped well into negative numbers, with zero at the middle of the graph. As best as I understand that statistic, a range, whether it's real, estimated, or rated, would never be below zero. I don't think this is related to any of the changes in that PR, but it's an interesting limit to consider.

**Comment by cwanja** [comment]
> Note in this image 

@tjhart unless I am missing it, your post does not include an image. 

> I displayed the data range on the right axis, (only for the purposes of testing and inspection), and to my surprise the scale for the estimated range dipped well into negative numbers, with zero at the middle of the graph. As best as I understand that statistic, a range, whether it's real, estimated, or rated, would never be below zero. I don't think this is related to any of the changes in that PR, but it's an interesting limit to consider.

Do you have a screenshot? Are you adding the ranges are axis markers? In all of the images here, only my [comment](https://github.com/teslamate-org/teslamate/issues/3763#issuecomment-2015943703) has them listed on the axis'.

**Comment by tjhart** [comment]
@cwanja Yeah, that's odd. I was trying to inline the last screen shot from @jheredianet 's post. Looked fine in `Preview` mode. Apologies for that. I've edited the comment.

I only added the axis so I could see the entire range being represented, as a means of inspection and testing. I was not trying to create a PR or suggest that those axis should be a part of the displayed graph.

**Comment by cwanja** [comment]
Perfect, thanks for that response and clarity @tjhart. Based upon that, @jheredianet I would not say that PR unintendedly resolves this issue. 

I would still like to see feedback from @Dulanic or @DrMichael before committing, but I am still thinking that removing `Standard options > Min 0` from Override 9 is the solution path.

**Comment by tjhart** [comment]
@cwanja 

Here's a screen shot with the range axis on the right. Again, this was just for me to visually see the scale. Zero seems to be vertically in the middle, which seems odd for these data sets.
Also, again, I expected one range for the distance data set, not two.
<img width="958" alt="Screenshot 2024-04-16 at 3 14 28 PM" src="https://github.com/teslamate-org/teslamate/assets/95165/f20dc709-0e8b-49f8-9366-d5f79956afaf">

**Comment by jheredianet** [comment]
> @cwanja
> 
> Here's a screen shot with the range axis on the right. Again, this was just for me to visually see the scale. Zero seems to be vertically in the middle, which seems odd for these data sets. Also, again, I expected one range for the distance data set, not two. <img alt="Screenshot 2024-04-16 at 3 14 28 PM" width="958" src="https://private-user-images.githubusercontent.com/95165/322986607-f20dc709-0e8b-49f8-9366-d5f79956afaf.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTMzMDMwMTksIm5iZiI6MTcxMzMwMjcxOSwicGF0aCI6Ii85NTE2NS8zMjI5ODY2MDctZjIwZGM3MDktMGU4Yi00OWY4LTkzNjYtZDVmNzk5NTZhZmFmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MTYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDE2VDIxMjUxOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTM5MTQxZjJiYmRiY2ZkZDA5Y2NjNWEyODMzNzlmNzliOGY3NzM0OTI3Y2RiYjgwODNiZjNhOWQ4ZWM0Njg0MTcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.KDqxvwtx-Z-jxitLnM1uyVaa3czxqqETG8GB8OzNqCs">

Here, we cannot think only in some metrics, we have to consider that in this graph there are other metrics together. So, the challenge is if we adapt the axis for a specific metric, it would not be well represented for other one. Having zero in the middle is special for the Power which is the one that has positive and negative values (and even elevation perhaps). Maybe the Solomonic solution could be separating the familiar metrics in another graph, to avoid confusion.

**Comment by cwanja** [comment]
> Maybe the Solomonic solution could be separating the familiar metrics in another graph, to avoid confusion.

I disagree with moving some of the metrics out to its own graph, the Drive Details is already busy enough. I do think if we can smooth out the lines where metrics collide in an "unnatural" way, I think that is the best method. I was unsuccessful at modifying my image to pull your merged PR #3836 and test in my installation. But my gut tells me we would still need further modifications to smooth the line and achieve what @tjhart is reporting. Awaiting a response here (or on the PR) on how I could test that specific PR.

**Comment by DrMichael** [comment]
> @DrMichael @Dulanic - can you review this thread and the proposed change below. To resolve the irregular lines spacing in the Drive Details panel, @tjhart proposed one of the following fixes (note the **orange** and **red** lines):
> 
> 1. Add `Standard options > Min 0` to Override 12
> 
> 2. Remove `Standard options > Min 0` from Override 9
> 
> In my testing, **option two** looks like the better path.
> 
> 1. Do you agree?
> 

Hmmm. just my gut feeling: I would prefer option one. Option 2 suggests. that the est range is less than half of the idel range...
